pkg/workflow/manager.go

```diff
@@ -223,6 +228,13 @@ func (w *WorkflowManager) syncWorkflow(key string) error {
 	if _, ok := workflow.Labels[workflowUID]; ok == false {
 		newWorkflow, err := w.setLabels(&workflow)
 		if err != nil {
+			if serr, ok := err.(*k8sApiErr.StatusError); ok {
```

Comment:
Standard execution path should not be indented, change to !ok { and handle the error